src/lib/components/ui/ArticleList.svelte

```diff
@@ -83,7 +97,17 @@
 	>
 		{#each filteredArticles.slice(0, visibleArticles) as article, index}
 			{#if index === visibleArticles - ARTICLES_PER_PAGE}
-				<div id={`article-${index}`} bind:this={newArticleRef}>
+				<div
+					id={`article-${index}`}
+					bind:this={newArticleRef}
+					style={article.is_sponsored
+						? [
+								`background-color: ${validateColor(article.sponsor_color ?? 'transparent')}`,
+								`color: ${validateColor(article.sponsor_text_color ?? 'inherit')}`
+							].join('; ')
+						: ''}
```

Comment:
```suggestion
					class={`bg-[${validateColor(...)}] text-[${validateColor(...)}]`}
```

Default to `class` instead of `style`